beluga-mqtt/src/manager.rs

```diff
@@ -0,0 +1,86 @@
+use std::collections::{HashMap, HashSet};
+
+use rumqttc::Publish;
+use tokio::sync::broadcast::{self, Receiver, Sender};
+
+#[derive(Debug, Default)]
+pub(super) struct SubscriberManager {
+    subscribed: HashMap<String, Sender<Publish>>,
+    unsubscribed: HashSet<String>,
+}
+
+impl SubscriberManager {
+    /// Returns a reference to the `Sender` for the given topic, if the
+    /// topic is currently subscribed.
+    pub(super) fn sender(&self, topic: &str) -> Option<&Sender<Publish>> {
+        self.subscribed.get(topic)
+    }
+
+    /// Returns a receiver for the given topic, if the topic is currently
+    /// subscribed
+    pub(super) fn receiver(&self, topic: &str) -> Option<Receiver<Publish>> {
+        self.subscribed.get(topic).map(|sender| sender.subscribe())
```

Comment:
I think at this point we need to keep track of the number of active subscribers to a given topic (basically reference counting).

Right now if we subscribe to topic "a" twice and then one of the two subscribers calls "unsubscribe" it will unsubscribe both of them.

Reply:
For now, you need to call unsubscribe on a client instance, subscriber only could receive a message. This question is more for an API design that we want at the end, let's discuss that.